website/static/js/fangorn.js

```diff
@@ -1013,6 +1161,448 @@ function scrollToFile(fileID) {
+                                rowButtons(),
+                                generalButtons
+                            ]
+                        )
+                    ])
+            ]);
+    }
+    if(tb.options.iconState.mode === 'search'){
+        return m('.row.tb-header-row', { 'data-mode' : 'search'},  [
+            m('#searchRow', { config : function () { $('#searchRow input').focus(); }}, [
+                        m('.col-xs-11',{ style : 'width: 90%'}, tb.options.filterTemplate.call(tb)),
+                        m('.col-xs-1',
+                            m('.fangorn-toolbar.pull-right',
+                                toolbarDismissIcon.call(tb)
+                            )
+                        )
+                    ])
+            ]);
+    }
+    if(tb.options.iconState.mode === 'rename'){
+        return m('.row.tb-header-row', [
+            m('#renameRow', { config : function () { $('#renameRow input').focus(); }}, [
+                        m('.col-xs-9', m('input#renameInput.tb-header-input', { value : tb.multiselected[0].data.name })),
+                        m('.col-xs-3.tb-buttons-col',
+                            m('.fangorn-toolbar.pull-right',
+                                [
+                                renameButton.call(tb),
+                                toolbarDismissIcon.call(tb)
+                                ]
+                            )
+                        )
+                    ])
+            ]);
+    }
+    if(tb.options.iconState.mode === 'createFolder'){
+        return m('.row.tb-header-row', [
+            m('#folderRow', { config : function () {
+                $('#folderRow input').focus();
+            }}, [
+                        m('.col-xs-9', [
+                            m('input#createFolderInput.tb-header-input', { placeholder : 'Folder name' }),
+                            m('#createFolderError.text-danger', { style : "display: none"})
+                            ]),
+                        m('.col-xs-3.tb-buttons-col',
+                            m('.fangorn-toolbar.pull-right',
+                                [
+                                createFolderButton.call(tb),
+                                toolbarDismissIcon.call(tb)
+                                ]
+                            )
+                        )
+                    ])
+            ]);
+    }
+}
+
+
+function _fangornResetToolbar () {
+    var tb = this;
+    if (tb.options.iconState.mode === 'search') {
+        tb.options.iconState = _defaultIconState();
+    }
+    tb.options.iconState.mode = 'bar';
+    tb.resetFilter();
+    m.redraw();
+}
+
+/**
+ * Toolbar icon templates
+ *
+ */
+function toolbarDismissIcon (){
+    var tb = this;
+    return m('.fangorn-toolbar-icon', {
+            onclick : function () {
+                _fangornResetToolbar.call(tb);
+            }
+        },
+        m('i.fa.fa-times')
+    );
+}
+ function searchIcon (){
```

Comment:
Remove unnecessary indent; it makes it difficult to tell if these functions are within a closure or are at the top-level scope.